### PR TITLE
Controller for file uploading and downloading

### PR DESCRIPTION
**Pull request by SemyonKorostik**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -15,6 +15,7 @@
     <description>English proficiency testing system</description>
     <properties>
         <java.version>11</java.version>
+        <security.version>5.5.1</security.version>
```
Comment (AntonMaksimenkau) [inline pom.xml]: Can the version be taken from the parent?

File: src/main/java/com/team4/testingsystem/controllers/FileController.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    public FileController(FileStorage fileStorage) {
+        this.fileStorage = fileStorage;
+    }
+
+    @GetMapping(path = "/download/{url}")
+    public ResponseEntity<Resource> download(@PathVariable("url") String url) {
+        return ResponseEntity.ok().body(fileStorage.load(url));
+    }
+
+    @PostMapping(path = "/upload")
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/controllers/FileController.java]: lets make path like '/files' with get mapping path '{url}' and postMapping empty path

File: src/main/java/com/team4/testingsystem/controllers/FileController.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    private final FileStorage fileStorage;
+
+    @Autowired
+    public FileController(FileStorage fileStorage) {
+        this.fileStorage = fileStorage;
+    }
+
+    @GetMapping(path = "/download/{url}")
+    public ResponseEntity<Resource> download(@PathVariable("url") String url) {
+        return ResponseEntity.ok().body(fileStorage.load(url));
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/controllers/FileController.java]: Do we really need response entities here?

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]

File: src/main/java/com/team4/testingsystem/controllers/FileController.java (hunk 3 above)
Comment (SemyonKorostik) [reply src/main/java/com/team4/testingsystem/controllers/FileController.java]: Probably, it would be more correct to return just a Resource

**Review by AntonMaksimenkau — APPROVED** [review]

File: src/test/java/com/team4/testingsystem/controllers/FileControllerIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    FileStorage fileStorage;
+
+    @Autowired
+    FileController fileController;
+
+    @Autowired
```
Comment (maxim12321) [inline src/test/java/com/team4/testingsystem/controllers/FileControllerIntegrationTest.java]: Shouldn't we use constructor injection here too?

**Review by maxim12321 — CHANGES_REQUESTED** [review]

**Review by maxim12321 — APPROVED** [review]

**Review by AntonMaksimenkau — APPROVED** [review]

File: src/main/java/com/team4/testingsystem/controllers/FileController.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        this.fileStorage = fileStorage;
+    }
+
+    @GetMapping(path = "/{url}")
+    public Resource download(@PathVariable("url") String url) {
+        return fileStorage.load(url);
+    }
+
+    @PostMapping(path = "/")
+    public String upload(@RequestParam MultipartFile file) {
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/controllers/FileController.java]: Don't need upload endpoint here

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]
See the comment

File: src/test/java/com/team4/testingsystem/controllers/FileControllerIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    @Test
+    void uploadSuccess() throws Exception {
+        MockMultipartFile mockMultipartFile = new MockMultipartFile("file", FILE_PATH,
+                MediaType.MULTIPART_FORM_DATA_VALUE, FILE_CONTENT.getBytes());
+
+        MockHttpServletRequestBuilder builder = multipart(PATH).file(mockMultipartFile)
+                .with(user(userDetails));
+
+        mockMvc.perform(builder).andExpect(status().isOk());
+
```
Comment (AntonMaksimenkau) [inline src/test/java/com/team4/testingsystem/controllers/FileControllerIntegrationTest.java:78]: Please, add a test for file not found

File: src/test/java/com/team4/testingsystem/controllers/FileControllerIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    void downloadInternalServerError() throws Exception {
+        saveFile();
+
+        mockMvc.perform(get(PATH + "{url}", WRONG_FILE_NAME)
+                .with(user(userDetails)))
+                .andExpect(status().isInternalServerError());
```
Comment (AntonMaksimenkau) [inline src/test/java/com/team4/testingsystem/controllers/FileControllerIntegrationTest.java:85]: it should be 404 here

File: src/test/java/com/team4/testingsystem/controllers/FileControllerIntegrationTest.java [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+        saveFile();
+
+        MockMultipartFile mockMultipartFile = new MockMultipartFile("file", FILE_PATH,
+                MediaType.MULTIPART_FORM_DATA_VALUE, FILE_CONTENT.getBytes());
+
+        MockHttpServletRequestBuilder builder = multipart(WRONG_PATH).file(mockMultipartFile)
+                .with(user(userDetails));
+
+        mockMvc.perform(builder).andExpect(status().isNotFound());
+    }
```
Comment (AntonMaksimenkau) [inline src/test/java/com/team4/testingsystem/controllers/FileControllerIntegrationTest.java:119]: What is the reason of such test?

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]

**Review by AntonMaksimenkau — APPROVED** [review]

**Pull request merged**